src/main/java/foundation/privacybydesign/sms/tokens/RedisTokenRequestRepository.java

```diff
@@ -117,4 +124,10 @@ public TokenRequest retrieve(String phone) {
             }
         }
     }
+
+    private static String CreateHash(String input) throws NoSuchAlgorithmException {
+        final MessageDigest digest = MessageDigest.getInstance("SHA-256");
+        final byte[] encodedhash = digest.digest(input.getBytes(StandardCharsets.UTF_8));
+        return Base64.getEncoder().encodeToString(encodedhash);
```

Comment:
Hashing phone numbers with plain SHA256 (instead of a keyed hash) does not serve any security purpose, as such hashes are trivial to reverse. Using a keyed hash with the key configured into the application only improves security if the attacker model is such that an attacker would not likely get simultaneous access to the application configuration if they get access to the database.

Reply:
Thank you for providing your feedback.

The reason for hashing the phonenumber (and soon also IP) in the key, is mainly to prevent any logging of keys with plaintext phonenumbers and IP-addresses somewhere. At least for now it is not directly readable, even if it is not fully secure yet. For now we accept the risk of external actors gaining access to our Redis instance.

Please feel free to raise an issue or create a pull-request, so we can work on this at a later moment.

Reply:
To be clear, it's all fine to me, and I agree that "for now we accept the risk of external actors gaining access to our Redis instance". However, I'd say you're better off not implementing any obfuscation at all. With this PR, the data _is_ still being logged as plaintext; just not plaintext a human can read immediately.